src/System.Console/tests/CancelKeyPress.Unix.cs

```diff
@@ -22,6 +23,41 @@ public void HandlerInvokedForSigQuit()
         HandlerInvokedForSignal(SIGQUIT);
     }
 
+    [PlatformSpecific(TestPlatforms.AnyUnix)]  // events are triggered by Unix signals (SIGINT, SIGQUIT, SIGCHLD).
+    public void ExitDetectionNotBlockedByHandler()
+    {
+        RemoteInvoke(() =>
+        {
+            var sempahore = new SemaphoreSlim(0, 1);
```

Comment:
SemaphoreSlim works, but given this usage, it seems like ManualResetEventSlim would be more directly appropriate.